rfc/0010-rancher-integration-policies.md

```diff
@@ -0,0 +1,162 @@
+|              |                                  |
+| :----------- | :------------------------------- |
+| Feature Name | Rancher integration of Kubewarden Policies  |
+| Start Date   | 2022-08-22                       |
+| Category     | Development                      |
+| RFC PR       | https://github.com/kubewarden/rfc/pull/11  |
+| State        | **ACCEPTED**                     |
+
+
+# Summary
+[summary]: #summary
+
+This RFC explains how the Kubewarden policies, either owned by the Kubewarden
+team or 3rd party ones, can be integrated in Rancher Explorer.
+
+For policies owned by the Kubewarden team  it proposes a development workflow
+using git branches and automation. The needed Rancher Helm charts and Rancher
+Packages artifacts metadata is tracked in a separate orphan branch for each
+artifact's repo, and submitted to https://github.com/rancher/charts.
+
+For 3rd party policies not owned by the Kubewarden team,
+it proposes using the Artifact Hub API to poll for the metadata stored in
+`policy/artifacthub-pkg.yml`.
+
+
+# Motivation
+[motivation]: #motivation
+
+Problems to solve:
+
+- Obtain the list of available policies regardless of where they come from, and
+  their metadata to show in Rancher Explorer UI.
+- Allow for groups of policies to be installed.
+- Implement it efficiently and with low development maintenance costs.
+- Allow for airgap installations.
+
+## Examples / User Stories
+[examples]: #examples
+
+2. As a user, I want to install one Kubewarden policy owned by the
+   Kubewarden/Rancher team via Rancher Explorer UI.
+3. As a user, I want to install several Kubewarden policies owned by the
+   Kubewarden/Rancher team via Rancher Explorer UI.
+4. As a user, I want to install 3rd party Kubewarden policies via Rancher
+   Explorer UI.
+5. As a Kubewarden developer, I want to release a new Kubewarden chart/policy so
+   it can be installed via Rancher Explorer UI.
+6. As a 3rd party developer, I want to release a policy on Artifact Hub so it can
+   be installed via Rancher Explorer UI.
+
+# Detailed design
+[design]: #detailed-design
+
+## For policies owned by the Kubewarden team
+
+Follow the process listed in [RFC-9, Rancher integration of Kubewarden
+charts](./0009-rancher-integration-charts.md).
+
+Carry the Helm chart code for the policy in a `rancher-X` orphan branch, where `X`
+is the release tag of the policy being targeted.
+Using these branches allows for easy rebases and cherry-picks, and documentating
+needed changes per commit. This also separates the Rancher vendored code, which
+may be of no interest to policy authors, and may coexist with other vendors.
```

Comment:
Wouldn't be simpler to just have it under a dedicated directory (like `rancher/`)?

Is the dedicated git branch an orphan one (meaning completely disconnected from `main`, à la `github-pages` so to speak)

Reply:
A folder could be fine.
One could put the needed Helm files in root too, and use `.helmignore` to filter the rest. This allows for reusing readme, license, and so.
But, if we want the UI to take care of the Helm Chart, then the chart should be `catalog.cattle.io/hidden: true` so we shouldn't need to care about anything of that, no need to show the readme, and all config is done via questions.yaml.

See https://github.com/kubewarden/allow-privilege-escalation-psp-policy/pull/30